src/extensions/default/NavigationAndHistory/main.js

```diff
@@ -0,0 +1,402 @@
+        var $link, $newItem;
+        $mrofContainer = $(htmlTemplate).appendTo("#editor-holder");
+        var $mrofList = $mrofContainer.find("#mrof-list");
+        
+        /**
+         * Focus handler for the link in list item 
+         * @private
+         */
+        function _onFocus(event) {
+            $("#mrof-container > #mrof-list > li.highlight").removeClass("highlight");
+            $(event.target).parent().addClass("highlight");
+            $mrofContainer.find("#recent-file-path").text($(event.target).parent().data("path"));
+            $currentContext = $(event.target).parent();
+        }
+        
+        /**
+         * Click handler for the link in list item 
+         * @private
+         */
+        function _onClick(event) {
+            var $context = $(event.delegateTarget).parent();
+            _openEditorForContext({
+                path: $context.data("path"),
+                paneId: $context.data("paneId"),
+                cursor: $context.data("cursor")
+            });
+        }
+        
+        /**
+         * Clears the MROF list in memory and pop over
+         * @private
+         */
+        function _clearMROFList() {
+            _mrofList = [];
+            $mrofList.empty();
+        }
+        
+        $("#mrof-list-close").one("click", _hideMROFList);
+        
+        var data, fileEntry;
+        
+        _syncWithFileSystem();
+        
+        // Iterate over the MROF list and create the pop over UI items
+        $.each(_mrofList, function (index, value) {
+            
+            data = {fullPath: value.file,
+                    name: FileUtils.getBaseName(value.file),
+                    isFile: true};
+            
+            fileEntry = FileSystem.getFileForPath(value.file);
+            
+            // Create new list item with a link
+            $link = $("<a href='#' class='mroitem'></a>").html(ViewUtils.getFileEntryDisplay({name: FileUtils.getBaseName(value.file)}));
+            
+            // Use the file icon providers
+            WorkingSetView.useIconProviders(data, $link);
+            
+            $newItem = $("<li></li>").append($link);
+            
+            $newItem.data("path", value.file);
+            $newItem.data("paneId", value.paneId);
+            $newItem.data("cursor", value.cursor);
+            $newItem.data("file", fileEntry);
+            $newItem.attr("title", value.file);
+            
+            // Use the class providers(git e.t.c)
+            WorkingSetView.useClassProviders(data, $newItem);
+            
+            if (_isOpenAndDirty(fileEntry)) {
+                $("<div class='file-status-icon dirty' style='position: static;float:left;margin-left: -5px;'></div>").prependTo($newItem);
+            }
+            
+            $mrofList.append($newItem);
+        });
+        
+        // Handlers for mouse events on the list items
+        $("#mrof-container > #mrof-list > li > a.mroitem").on("focus", _onFocus);
```

Comment:
Reused reference `$("#mrof-container > #mrof-list > li > a.mroitem")` can be again set to variable